Chore: avoid misleading "Unpermitted parameters: xxx" message log

cbliard opened the pull request:
When `PermittedParams#settings` was called, it was not permitting any params, which lead to "Unpermitted parameters: ..." messages being logged, and then merged the resulting empty params with the controller params after having filtered them manually. It was ok-ish but the logged error message was misleading.

The `#settings` method was refactored to permit params that are allowed in a single step and avoid the misleading log messages.

The rubocop linting has been put in a separate commit for easier reviewing.

cbliard committed:
Use double-quotes and make Rubocop happier

cbliard committed:
Avoid misleading "Unpermitted parameters: xxx" log

When `PermittedParams#settings` was called, it was not permitting any
params, which lead to "Unpermitted parameters: ..." messages being
logged, and then merged the resulting empty params with the controller
params after having filtered them manually. It was ok-ish but the logged
error message was misleading.

The `#settings` method was refactored to permit params that are allowed
in a single step and avoid the misleading log messages.

klaustopher commented on app/models/permitted_params/allowed_settings.rb:
```diff
@@ -29,6 +29,31 @@ def all
+      Settings::Definition.all.each do |key, definition| # rubocop:disable Rails/FindEach
+        next if restricted_keys.include?(key)
+
+        case definition.format
+        when :hash
+          complex_filters[key] = {}
+        when :array
+          complex_filters[key] = []
+        else
+          scalar_filters << key
+        end
```
From the docs, the only argument for [`ActionController::Parameters#permit`](https://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-permit) is a list of filters, no kwargs. So we would not need to seperate scalar and complex filters and could simply rewrite this to:

```suggestion
        case definition.format
        when :hash
          scalar_filters << { key => {} }
        when :array
          scalar_filters << { key => [] }
        else
          scalar_filters << key
        end
```

cbliard committed:
refactor: simplify filters for permitted params

`ActionController::Parameters#permit` takes a list of filters. There is
no kwargs argument.

Co-authored-by: Klaus Zanders <k.zanders@openproject.com>

cbliard committed:
Allow to permit additional parameters in `PermittedParams#settings`

"attachment_whitelist" is normally permitted as an array parameter, but
the form passes it as a string parameter to be split. This commit adds a
way to also permit parameters such as "attachment_whitelist" as a string
parameter.
